commit 0ded2a069f270507a6c8020688582a47babbcc83
Author: David Faure <faure@kde.org>
Date:   Wed Jun 18 20:29:53 2014 +0200

    Handle unexpected return values here (such as -1).
    
    Patch by Jim Scadden, jims at silentasylum.co.uk

diff --git a/kio/kio/tcpslavebase.cpp b/kio/kio/tcpslavebase.cpp
index 2af99bf..cdf28f0 100644
--- a/kio/kio/tcpslavebase.cpp
+++ b/kio/kio/tcpslavebase.cpp
@@ -824,6 +824,9 @@ TCPSlaveBase::SslResult TCPSlaveBase::verifyServerCertificate()
             messageBox(SSLMessageBox /*the SSL info dialog*/, d->host);
         } else if (msgResult == KMessageBox::Cancel) {
             return ResultFailed;
+        } else if (msgResult != KMessageBox::No) {
+            kWarning() << "Unexpected MessageBox response received:" << msgResult;
+            return ResultFailed;
         }
         //fall through on KMessageBox::No
     } while (msgResult == KMessageBox::Yes);
